internal/bot/bot.go

```diff
@@ -24,9 +25,10 @@ func Start() {
 	go commands.StartSubscriber()
 
 	// Add the endpoint for github webhook payloads
-	http.HandleFunc("/github", github.WebhookListener)
+	router := httprouter.New()
+	router.POST("/github/:channelID", github.WebhookListener)
 
 	// Start the HTTP server ()
 	address := os.Getenv("address")
-	logrus.Fatal(http.ListenAndServe(address, nil))
+	logrus.Fatal(http.ListenAndServe(address, router))
```

Comment:
The default `ListenAndServe` method uses the default HTTP server. Some of the default values are set to extremely long values or are set to 0 (which disables them). Cloudflare has written an extensive blog post on how to set up a production-ready Go HTTP server and HTTP client. https://blog.cloudflare.com/the-complete-guide-to-golang-net-http-timeouts/

TL;DR:
```go
srv := &http.Server{
    ReadTimeout: 5 * time.Second,
    WriteTimeout: 10 * time.Second,
}

err := srv.ListenAndServe(address, router)
if err != nil && err != http.ErrServerClosed {
  logrus.Fatal(err)
}
```

Reply:
Finally got around to fixing this :sweat_smile:, thanks for the tip!